cmd/status/cmdstatus.go

```diff
@@ -66,49 +95,167 @@ type Runner struct {
 	timeout   time.Duration
 	output    string
 
-	pollerFactoryFunc func(cmdutil.Factory) (poller.Poller, error)
+	invType          string
+	inventoryNames   string
+	inventoryNameSet map[string]bool
+	namespaces       string
+	namespaceSet     map[string]bool
+	statuses         string
+	statusSet        map[string]bool
+
+	PollerFactoryFunc func(cmdutil.Factory) (poller.Poller, error)
 }
 
-// runE implements the logic of the command and will delegate to the
-// poller to compute status for each of the resources. One of the printer
-// implementations takes care of printing the output.
-func (r *Runner) runE(cmd *cobra.Command, args []string) error {
+func (r *Runner) preRunE(*cobra.Command, []string) error {
+	if !slice.ContainsString(PollUntilOptions, r.pollUntil, nil) {
+		return fmt.Errorf("pollUntil must be one of %s, %s, %s, %s",
```

Comment:
Nit: I think it would be easier to just use `strings.Join(PollUntilOptions, ",")` here